tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -464,6 +464,99 @@ class SerializableModelLibraryGenerator {
     });
   }
 
+  Expression _buildClonableCopyWithAssignment(
+      SerializableModelFieldDefinition field) {
+    var shouldDeepClone = field.type.isListType || field.type.isMapType;
+
+    if (!shouldDeepClone) {
+      return refer(cloneTrivialFunctionName, serverpodUrl(serverCode))
+          .call([refer(Keyword.thisKeyword).property(field.name)]);
+    }
+
+    return _buildDeepCloneTree(
+      field.type,
+      field.name,
+      root: true,
+    );
+  }
+
+  Expression _buildDeepCloneTree(TypeDefinition type, String variableName,
+      {int depth = 0, bool root = false}) {
+    var isLeafNode = type.generics.isEmpty;
```

Comment:
This should be robust, but somehow looks a bit off to my eye. Maybe there is a cleaner check? I think this should be equal:
```dart
var isLeafNode = !(field.type.isListType || field.type.isMapType);
```

Reply:
I think this should be fine, but we might encounter issues if we ever allow "records" for example.